language-detection-fasttext/src/main/java/com/hpe/caf/languagedetection/fasttext/LangEncoding.java

```diff
@@ -0,0 +1,229 @@
+    HAUSA("ha"),
+    HEBREW("he"),
+    HERERO("hz"),
+    HINDI("hi"),
+    HIRI_MOTU("ho"),
+    HUNGARIAN("hu"),
+    INTERLINGUA("ia"),
+    INDONESIAN("id"),
+    INTERLINGUE("ie"),
+    IRISH("ga"),
+    IGBO("ig"),
+    INUPIAQ("ik"),
+    IDO("io"),
+    ICELANDIC("is"),
+    ITALIAN("it"),
+    INUKTITUT("iu"),
+    JAPANESE("ja"),
+    JAVANESE("jv"),
+    GREENLANDIC("kl"),
+    KANNADA("kn"),
+    KANURI("kr"),
+    KASHMIRI("ks"),
+    KAZAKH("kk"),
+    CENTRAL_KHMER("km"),
+    KIKUYU("ki"),
+    KINYARWANDA("rw"),
+    KIRGHIZ("ky"),
+    KOMI("kv"),
+    KONGO("kg"),
+    KOREAN("ko"),
+    KURDISH("ku"),
+    KUANYAMA("kj"),
+    LATIN("la"),
+    LUXEMBOURGISH("lb"),
+    GANDA("lg"),
+    LIMBURGAN("li"),
+    LINGALA("ln"),
+    LAO("lo"),
+    LITHUANIAN("lt"),
+    LUBA_KATANGA("lu"),
+    LATVIAN("lv"),
+    MANX("gv"),
+    MACEDONIAN("mk"),
+    MALAGASY("mg"),
+    MALAY("ms"),
+    MALAYALAM("ml"),
+    MALTESE("mt"),
+    MAORI("mi"),
+    MARATHI("mr"),
+    MARSHALLESE("mh"),
+    MONGOLIAN("mn"),
+    NAURU("na"),
+    NAVAJO("nv"),
+    NORTH_NDEBELE("nd"),
+    NEPALI("ne"),
+    NDONGA("ng"),
+    NORWEGIAN_BOKMÅL("nb"),
+    NORWEGIAN_NYNORSK("nn"),
+    NORWEGIAN("no"),
+    SICHUAN_YI("ii"),
+    SOUTH_NDEBELE("nr"),
+    OCCITAN("oc"),
+    OJIBWA("oj"),
+    OLD_SLAVONIC("cu"),
+    OROMO("om"),
+    ORIYA("or"),
+    OSSETIAN("os"),
+    PUNJABI("pa"),
+    PALI("pi"),
+    PERSIAN("fa"),
+    POLISH("pl"),
+    PASHTO("ps"),
+    PORTUGUESE("pt"),
+    QUECHUA("qu"),
+    ROMANSH("rm"),
+    RUNDI("rn"),
+    ROMANIAN("ro"),
+    RUSSIAN("ru"),
+    SANSKRIT("sa"),
+    SARDINIAN("sc"),
+    SINDHI("sd"),
+    NORTHERN_SAMI("se"),
+    SAMOAN("sm"),
+    SANGO("sg"),
+    SERBIAN("sr"),
+    SERBO_CROATIAN("sh"),
+    GAELIC("gd"),
+    SHONA("sn"),
+    SINHALA("si"),
+    SLOVAK("sk"),
+    SLOVENIAN("sl"),
+    SOMALI("so"),
+    SOUTHERN_SOTHO("st"),
+    SPANISH("es"),
+    SUNDANESE("su"),
+    SWAHILI("sw"),
+    SWATI("ss"),
+    SWEDISH("sv"),
+    TAMIL("ta"),
+    TELUGU("te"),
+    TAJIK("tg"),
+    THAI("th"),
+    TIGRINYA("ti"),
+    TIBETAN("bo"),
+    TURKMEN("tk"),
+    TAGALOG("tl"),
+    TSWANA("tn"),
+    TONGA("to"),
+    TURKISH("tr"),
+    TSONGA("ts"),
+    TATAR("tt"),
+    TWI("tw"),
+    TAHITIAN("ty"),
+    UIGHUR("ug"),
+    UKRAINIAN("uk"),
+    URDU("ur"),
+    UZBEK("uz"),
+    VENDA("ve"),
+    VIETNAMESE("vi"),
+    VOLAPÜK("vo"),
+    WALLOON("wa"),
+    WELSH("cy"),
+    WOLOF("wo"),
+    WESTERN_FRISIAN("fy"),
+    XHOSA("xh"),
+    YIDDISH("yi"),
+    YORUBA("yo"),
+    ZHUANG("za"),
+    ZULU("zu"),
+    UNKNOWN("unknown"); // added value to specify language with low detection accuracy
+
+    private final String langCode;
+    
+    private LangEncoding(final String langCode)
+    {
+        this.langCode = langCode;
+    }
+
+    public String getLangCode()
+    {
+        return langCode;
+    }
+
+    public static String getLangNameFromCode(final String code)
```

Comment:
Instead of this method could we use LangEncoding.valueOf(code).name();

Reply:
LangEncoding.valueOf(code).name() would return null, code would have be languageName.